### ISSUE TITLE
Implement Path Loss for Sionna physical environment

### ISSUE THREAD

**Issue by AshFungor** [CAVISE/artery#59]
We need to implement INET's pass loss, but with Sionna's solvers:

```cpp
class INET_API IPathLoss : public IPrintableObject
{
  public:
    /**
     * Returns the loss factor for the provided transmission and arrival.
     * The value is in the range [0, 1] where 1 means no loss at all and 0
     * means all power is lost.
     */
    virtual double computePathLoss(const ITransmission *transmission, const IArrival *arrival) const = 0;

    /**
     * Returns the loss factor as a function of propagation speed, carrier
     * frequency and distance. The value is in the range [0, 1] where 1 means
     * no loss at all and 0 means all power is lost.
     */
    virtual double computePathLoss(mps propagationSpeed, Hz frequency, m distance) const = 0;

    /**
     * Returns the range for the given loss factor. The value is in the range
     * [0, +infinity) or NaN if unspecified.
     */
    virtual m computeRange(mps propagationSpeed, Hz frequency, double loss) const = 0;
};
```

I'm not sure we will be able to port all of this, since, for example, carrier frequency is set globally for all transmitters and receivers in Sionna. As an example, GemV2: 

```cpp
double PathLoss::computePathLoss(const phy::ITransmission* transmission, const phy::IArrival* arrival) const
{
    inet::Coord tx = transmission->getStartPosition();
    inet::Coord rx = arrival->getStartPosition();

    LinkClass link = m_classifier->classifyLink(Position { tx.x, tx.y }, Position { rx.x, rx.y });
    IPathLoss* model = nullptr;
    meter range { 0.0 };
    switch (link)
    {
        case LinkClass::LOS:
            model = m_los;
            range = m_range_los;
            EV_DETAIL << "LOS propagation for " << *transmission << "\n";
            break;
        case LinkClass::NLOSb:
            model = m_nlos_b;
            range = m_range_nlos_b;
            EV_DETAIL << "NLOSb propagation for " << *transmission << "\n";
            break;
        case LinkClass::NLOSf:
            model = m_nlos_f;
            range = m_range_nlos_f;
            EV_DETAIL << "NLOSf propagation for " << *transmission << "\n";
            break;
        case LinkClass::NLOSv:
            model = m_nlos_v;
            range = m_range_nlos_v;
            EV_DETAIL << "NLOSv propagation for " << *transmission << "\n";
            break;
        default:
            throw cRuntimeError("invalid link classification");
    };

    // compare model's maximum range with actual distance
    if (tx.distance(rx) > range.get()) {
        return 0.0; // all signal power is lost
    }

    double loss = model->computePathLoss(transmission, arrival);
    if (m_small_scale) {
        loss *= m_small_scale->computeVariation(Position { tx.x, tx.y }, Position { rx.x, rx.y }, range, link);
    }
    return loss;
}
```

We would also probably have to run all solvers in batch and provide visualizations, so any improvements like this are also welcome.
